src/runtime/nitro.ts

```diff
@@ -1,12 +1,132 @@
+import path from 'node:path'
+import crypto from 'node:crypto'
 import type { NitroAppPlugin, RenderResponse } from 'nitropack'
+import type { H3Event } from 'h3'
+import { useRuntimeConfig } from '#imports'
+import type {
+  ModuleOptions,
+  ContentSecurityPolicyValue,
+  SecurityHeaders,
+  MiddlewareConfiguration
+} from '../types'
+
+interface NuxtRenderHTMLContext {
+  island?: boolean
+  htmlAttrs: string[]
+  head: string[]
+  bodyAttrs: string[]
+  bodyPrepend: string[]
+  body: string[]
+  bodyAppend: string[]
+}
 
 export default <NitroAppPlugin> function (nitro) {
-  nitro.hooks.hook('render:response', (response: RenderResponse) => {
+  nitro.hooks.hook('render:response', (response: RenderResponse, { event }: { event: H3Event }) => {
     // Temporary as in Nuxt 3.0.0 header name is 'X-Powered-By' and in 3.1.X is 'x-powered-by'
     if (response.headers['x-powered-by']) {
       delete response.headers['x-powered-by']
     } else if (response.headers['X-Powered-By']) {
       delete response.headers['X-Powered-By']
     }
   })
+
+  nitro.hooks.hook('render:html', (html: NuxtRenderHTMLContext, { event }: { event: H3Event }) => {
```

Comment:
I would recommend creating a new file where this custom plugin could be registered as currently it is registered as a part of a XPoweredBy plugin so once a user disabled hidePoweredBy, this hook will not be triggered as well.

What is more, I would enable this module without any configuration if the project is running in SSG by default and  remain configurability from `security.headers.contentSecurityPolicy.value`